store/src/sqlite/model.rs

```diff
@@ -0,0 +1,1184 @@
+            "SELECT
+               TOTAL(ahash_0) & 0xFFFFFFFF, TOTAL(ahash_1) & 0xFFFFFFFF,
+               TOTAL(ahash_2) & 0xFFFFFFFF, TOTAL(ahash_3) & 0xFFFFFFFF,
+               TOTAL(ahash_4) & 0xFFFFFFFF, TOTAL(ahash_5) & 0xFFFFFFFF,
+               TOTAL(ahash_6) & 0xFFFFFFFF, TOTAL(ahash_7) & 0xFFFFFFFF,
+               COUNT(1)
+             FROM model_key WHERE key > ? AND key < ?;",
+        );
+        let row = query
+            .bind(left_fencepost.as_bytes())
+            .bind(right_fencepost.as_bytes())
+            .fetch_one(self.pool.reader())
+            .await?;
+        let bytes: [u32; 8] = [
+            row.get(0),
+            row.get(1),
+            row.get(2),
+            row.get(3),
+            row.get(4),
+            row.get(5),
+            row.get(6),
+            row.get(7),
+        ];
+        let count: i64 = row.get(8); // sql int type is signed
+        let count: u64 = count
+            .try_into()
+            .expect("COUNT(1) should never return a negative number");
+        Ok(HashCount::new(H::from(bytes), count))
+    }
+
+    #[instrument(skip(self))]
+    async fn range(
+        &mut self,
+        left_fencepost: &Self::Key,
+        right_fencepost: &Self::Key,
+        offset: usize,
+        limit: usize,
+    ) -> Result<Box<dyn Iterator<Item = Self::Key> + Send + 'static>> {
+        let query = sqlx::query(
+            "
+        SELECT
+            key
+        FROM
+            model_key
+        WHERE
+            key > ? AND key < ?
+        ORDER BY
+            key ASC
+        LIMIT
+            ?
+        OFFSET
+            ?;
+        ",
+        );
+        let rows = query
+            .bind(left_fencepost.as_bytes())
+            .bind(right_fencepost.as_bytes())
+            .bind(limit as i64)
+            .bind(offset as i64)
+            .fetch_all(self.pool.reader())
+            .await?;
+        //debug!(count = rows.len(), "rows");
+        Ok(Box::new(rows.into_iter().map(|row| {
+            let bytes: Vec<u8> = row.get(0);
+            EventId::from(bytes)
+        })))
+    }
+    #[instrument(skip(self))]
+    async fn range_with_values(
+        &mut self,
+        left_fencepost: &Self::Key,
+        right_fencepost: &Self::Key,
+        offset: usize,
+        limit: usize,
+    ) -> Result<Box<dyn Iterator<Item = (Self::Key, Vec<u8>)> + Send + 'static>> {
+        let query = sqlx::query(
+            "
+        SELECT
+            model_block.key, model_block.cid, model_block.root, model_block.bytes
+        FROM (
+            SELECT
+                key
+            FROM model_key
+            WHERE
+                key > ? AND key < ?
+                AND value_retrieved = true
+            ORDER BY
+                key ASC
+            LIMIT
+                ?
+            OFFSET
+                ?
+        ) key
+        JOIN
+            model_block
+        ON
+            key.key = model_block.key
+        ;",
```

Comment:
I expect this query is relying on the model_block order index. If not we should likely find a better index.

Reply:
yea, this one was interesting! already told you on discord, but when I changed the PK to `(key, cid)` it changed the order of these results and resulted in a different CAR file. added an explicit order by `ORDER BY model_block.key, model_block.idx` and we're good. 

it should be using indexes, we have two on model_block `(key, cid)` and `(cid)`, and the key should be sufficient here (can do some EXPLAIN but I'll need to build another db with data in it). model_key has `(key, value_retrieved)` and `(key)` so I think we have good index coverage on all the predicates.